packages/react/collection/src/Collection.tsx

```diff
@@ -96,18 +96,19 @@ function createCollection<ItemElement extends HTMLElement, ItemData>() {
 
   function useCollection() {
     const context = React.useContext(Context);
-    return {
-      getItems() {
-        const orderedNodes = Array.from(
-          context.collectionRef.current!.querySelectorAll(`[${ITEM_DATA_ATTR}]`)
-        );
-        const items = Array.from(context.itemMap.values());
-        const orderedItems = items.sort(
-          (a, b) => orderedNodes.indexOf(a.ref.current!) - orderedNodes.indexOf(b.ref.current!)
-        );
-        return orderedItems;
-      },
-    };
+
+    const getItems = React.useCallback(() => {
+      const orderedNodes = Array.from(
+        context.collectionRef.current!.querySelectorAll(`[${ITEM_DATA_ATTR}]`)
+      );
+      const items = Array.from(context.itemMap.values());
+      const orderedItems = items.sort(
+        (a, b) => orderedNodes.indexOf(a.ref.current!) - orderedNodes.indexOf(b.ref.current!)
+      );
+      return orderedItems;
+    }, [context.collectionRef, context.itemMap]);
+
+    return { getItems };
```

Comment:
I'm wondering if it's worth simplifying the API as well now that we're here.
The hook only returns one thing, making the function the return itself would make more sense then? What do you reckon?

Reply:
Yeah, was thinking the same